### PR TITLE
mpi: add all info generated by mpi-info.c

### PR DESCRIPTION
**Pull request by AntonLydike**
This PR uses the `mpi-info.c` file in #526 to generate the rest of the library info object.

### CODE REVIEW

**Review by webmiche — APPROVED** [review]
Personally, I prefer to have these kinds of magic numbers in hex. Feels more intuitive when looking at them...

Other than that, this is fine with me :+1:

File: xdsl/transforms/lower_mpi.py [hunk 1]
```diff
@@ -25,29 +25,65 @@ class MpiLibraryInfo:
 
     This way of doing it is inherently fragile, but we don't know of any better way.
     We plan to include a C file that automagically extracts all this information from MPI headers.
+    You can see the current C file used in this PR: https://github.com/xdslproject/xdsl/pull/526
+    You can see the status of OpenMPI support here: https://github.com/xdslproject/xdsl/issues/523
 
-    These defaults have been chosen to work with **our** version of OpenMPI. No guarantees of portability!
+    These defaults have been chosen to work with a specific version of MPICH. We would suggest
```
Comment (georgebisbas) [inline xdsl/transforms/lower_mpi.py:31]: You can add which version

**Review by georgebisbas — APPROVED** [review]

**Review by superlopuh — APPROVED** [review]

File: xdsl/transforms/lower_mpi.py (hunk 1 above)
Comment (AntonLydike) [reply xdsl/transforms/lower_mpi.py]: Added the version!

**Comment by AntonLydike** [comment]
I changed the values to hex and added the version comment. If everything is fine with you I'll merge this now!

**Pull request merged**